examples/http_headers/src/lib.rs

```diff
@@ -43,10 +74,25 @@ impl Context for HttpHeaders {}
 
 impl HttpContext for HttpHeaders {
     fn on_http_request_headers(&mut self, _: usize, _: bool) -> Action {
+        
+        let lookup = LookupFuture::lookup(
+            vec![Name::from_str("www.example.com.").unwrap()],
+            RecordType::A,
+            Default::default(),
+            CachingClient::new(0, client, false),
+        );
+    
+        let io_loop = Runtime::new().unwrap();
+        let lookup = io_loop.block_on(lookup).unwrap();
```

Comment:
This seems like a bad idea, since each worker thread (in case of Envoy) is going to be blocked on this, and DNS resolution can potentially take hundreds of milliseconds... not to mention the overhead of starting Tokio runtime per each request.

Reply:
Can we have a meeting about this? Can we mitigate the overhead per request by cache layer?

Reply:
You could store the runtime in a global variable and share it across the callbacks, but I suspect that Tokio runtime couldn't make any progress without deep integration with Proxy-Wasm's callbacks.

Also, if you try to compile your example, you can see that Tokio async doesn't work in Wasm (yet):
```
compile_error!("Only features sync,macros,io-util,rt,time are supported on wasm.");
```
What are you trying to achieve? What's your end goal?

Reply:
yup... I can  not compile through for that reason. I am trying to get cname chain in WASM. Could you recommend some feasible way? I saw there are some other lib besides Tokio, RustDNS but not sure if it has similar issue. It will be nice if you can suggest

Reply:
Are you only interested in the final answer or the whole CNAME chain? If the former, and you're using Envoy, then there are `envoy_resolve_dns` / `envoy_on_resolve_dns` extensions that you could call and use Envoy's built-in DNS resolver... Note that they are not exposed in the Rust SDK yet.